src/averageAges.js

```diff
@@ -37,8 +43,20 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
-}
+  const filteredWomen = people.filter(person => {
+    if (withChildren) {
+      return people.some(
+        child => child.mother === person.name) && person.sex === 'f';
+    } else {
+      return person.sex === 'f';
+    }
+  });
```

Comment:
In task is said to not to use `if-else`

you can do something similar as you do in `calculateMenAverageAge`